### PR TITLE
fix: normalize `#build` windows path for dev and prerender presets

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/4232

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The fix applied to dev preset also needs to be applied to prerender preset. It seemed easiest way was to extract from options and add to preset itself. There may be a better way to do this.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/presets/nitro-dev.ts [hunk 1]
```diff
@@ -8,5 +10,13 @@ export const nitroDev = defineNitroPreset({
   },
   externals: { trace: false },
   inlineDynamicImports: true, // externals plugin limitation
-  sourceMap: true
+  sourceMap: true,
+  hooks: {
+    'nitro:rollup:before' (nitro) {
+      if (isWindows) {
+        // Windows dynamic imports should be file:// url
+        nitro.options.alias['#build'] = pathToFileURL(nitro.options.buildDir).href
```
Comment (pi0) [inline src/presets/nitro-dev.ts:18]: Shared situation with both is `externals: { trace: false }`

File: src/presets/nitro-prerender.ts [hunk 2]
```diff
@@ -6,5 +8,13 @@ export const nitroPrerender = defineNitroPreset({
   output: {
     serverDir: '{{ buildDir }}/prerender'
   },
-  externals: { trace: false }
+  externals: { trace: false },
+  hooks: {
+    'nitro:rollup:before' (nitro) {
+      if (isWindows) {
+      // Windows dynamic imports should be file:// url
```
Comment (pi0) [inline src/presets/nitro-prerender.ts:15]:
```suggestion
       // Windows dynamic imports should be file:// url
```

**Review by pi0 — APPROVED** [review]
LGTM. Seems a more clear workaround at least.

File: src/presets/nitro-dev.ts (hunk 1 above)
Comment (danielroe) [reply src/presets/nitro-dev.ts:18]: Hm. Perhaps we could just test for that instead in rollup config, or are you happy with the preset option?

File: src/presets/nitro-prerender.ts [hunk 3]
```diff
@@ -6,5 +8,13 @@ export const nitroPrerender = defineNitroPreset({
   output: {
     serverDir: '{{ buildDir }}/prerender'
   },
-  externals: { trace: false }
+  externals: { trace: false },
+  hooks: {
+    'nitro:rollup:before' (nitro) {
+      if (isWindows) {
+       // Windows dynamic imports should be file:// url
```
Comment (pi0) [inline src/presets/nitro-prerender.ts:15]:
```suggestion
        // Windows dynamic imports should be file:// url
```

File: src/presets/nitro-dev.ts (hunk 1 above)
Comment (pi0) [reply src/presets/nitro-dev.ts:18]: Improved in https://github.com/unjs/nitro/commit/1ec2ee4cf3dfd355a2b7d8d6d7aa5de6dcb51b7c. Extending alias was too late here.

**Pull request merged**